pkg/controller/vitesskeyspace/reconcile_resharding.go

```diff
@@ -133,6 +133,7 @@ func (r *reconcileHandler) reconcileResharding(ctx context.Context) (reconcile.R
 	}
 
 	sort.Strings(workflowStatus.ServingShards)
+	r.vtk.Status.Resharding = *workflowStatus
```

Comment:
Rather than deref, let's just make the field type a pointer type to avoid copy semantics.

Reply:
Done.